app/controllers/v1/terms_controller.rb

```diff
@@ -0,0 +1,123 @@
+module V1
+  class TermsController < ApplicationController
+    # Check that vocabulary is valid, if not should render and error
+
+    ## force json type
+
+    # GET /vocabularies/:string_key/terms
+    def index
+      if valid_search_params?
+        solr_response = URIService.solr.search do |solr_params|
+          solr_params.vocabulary params[:vocabulary_string_key]
+          solr_params.q          params[:q]
+          solr_params.authority  params[:authority]
+          solr_params.uri        params[:uri]
+          solr_params.pref_label params[:pref_label]
+          solr_params.alt_label  params[:alt_label]
+          solr_params.term_type  params[:term_type]
+
+          custom_fields.each do |k, v|
+            if params[k]
+              solr_params.fq("#{k}#{URIService.solr_suffix(v[:data_type])}", params[k])
+            end
+          end
+        end
+
+        response = URIService::JSON.term_search(solr_response)
+      else
+        response = { page: 1, per_page: 25, total_records: 0, terms: [] }
+      end
+
+      render json: response, status: 200
+    end
+
+    # GET /vocabularies/:string_key/terms/:uri
+    def show
+      if (term = URIService.solr.find_term(params[:vocabulary_string_key], params[:uri]))
+        render json: URIService::JSON.term(term), status: 200
+      else
+        render json: URIService::JSON.errors('Not Found'), status: 404
+      end
+    end
+
+    # POST /vocabularies/:string_key/terms
+    def create
+      term = Term.new(create_params)
+      term.vocabulary = vocabulary
+
+      custom_fields.each do |f, v|
+        term.add_custom_field(f, params.fetch(f, nil))
+      end
+
+      if term.save
+        render json: URIService::JSON.term(term), status: 201
+      else
+        render json: URIService::JSON.errors(term.errors.full_messages), status: 400
+      end
+    end
+
+    # PATCH /vocabularies/:string_key/terms/:uri
+    def update
+      term = Term.find_by(vocabulary: vocabulary, uri: params[:uri]) # should include querying for vocab
+
+      if term.nil?
+        render json: URIService::JSON.errors('Not Found'), status: 404
+      else
+        term.assign_attributes(update_params) # updates, but doesn't save.
+
+        custom_fields.each do |f, v|
+          term.add_custom_field(f, params.fetch(f, nil))
```

Comment:
I think you have a good point there. It would clear out the value. I think I was trying to allow for the clearing of values if a user sends them through as null. But I can check for the presence of the value in the params hash before setting the custom field and that should fix the problem. I'll add tests for this too.

Reply:
Perfect, thanks!

Reply:
Added a commit that fixes this and added a test for it.